tesseract/consensus/polygon-pos/src/lib.rs

```diff
@@ -1,122 +1,183 @@
+	pub async fn get_consensus_state(&self) -> Result<ConsensusState, anyhow::Error> {
+		let consensus_state_serialized: Vec<u8> =
+			self.provider.query_consensus_state(None, self.consensus_state_id).await?;
+		Ok(ConsensusState::decode(&mut &consensus_state_serialized[..])?)
+	}
 
-		Ok(consensus_state)
+	/// Get the ISMP provider
+	pub fn provider(&self) -> Arc<dyn IsmpProvider> {
+		self.provider.clone()
 	}
 }
 
-pub struct KeccakHasher;
+pub const POLYGON_CONSENSUS_CLIENT_ID: ConsensusClientId = *b"PLGN";
 
+/// Keccak hasher for EVM headers
+pub struct KeccakHasher;
 impl Keccak256 for KeccakHasher {
-	fn keccak256(bytes: &[u8]) -> primitive_types::H256
-	where
-		Self: Sized,
-	{
+	fn keccak256(bytes: &[u8]) -> primitive_types::H256 {
 		sp_core::keccak_256(bytes).into()
 	}
 }
+
+#[async_trait::async_trait]
+impl IsmpHost for PolygonPosHost {
+	async fn start_consensus(
+		&self,
+		counterparty: Arc<dyn IsmpProvider>,
+	) -> Result<(), anyhow::Error> {
+		use crate::notification::consensus_notification;
+		let interval = tokio::time::interval(Duration::from_secs(
+			self.host.consensus_update_frequency.unwrap_or(300),
+		));
+		let client = self.clone();
+		let counterparty_clone = counterparty.clone();
+		let mut interval = Box::pin(interval);
+		let provider = self.provider();
+		loop {
+			interval.as_mut().tick().await;
+			match consensus_notification(&client, counterparty_clone.clone()).await {
+				Ok(Some(update)) => {
+					use ismp::messaging::ConsensusMessage;
+					let consensus_message = ConsensusMessage {
+						consensus_proof: update.encode(),
+						consensus_state_id: client.consensus_state_id,
+						signer: vec![], // Who is the signer?
+					};
+					log::info!(
+						target: "tesseract",
+						"🛰️ Transmitting consensus message from {} to {}",
+						provider.name(), counterparty.name()
+					);
+					let res = counterparty
+						.submit(
+							vec![Message::Consensus(consensus_message)],
+							counterparty.state_machine_id().state_id,
+						)
+						.await;
+					if let Err(err) = res {
+						log::error!(
+							"Failed to submit transaction to {}: {err:?}",
+							counterparty.name()
+						)
+					}
+				},
+				Ok(None) => {
+					// No update to send, just continue
+				},
+				Err(e) => {
+					log::error!(target: "tesseract","Consensus task {}->{} encountered an error: {e:?}", provider.name(), counterparty.name())
+				},
+			}
+		}
+	}
+
+	async fn query_initial_consensus_state(
+		&self,
+	) -> Result<Option<CreateConsensusState>, anyhow::Error> {
+		let initial_consensus_state = self.get_consensus_state().await.map_err(|e| {
+			anyhow::anyhow!("PolygonPosHost: fetch initial consensus state failed: {e}")
+		})?;
+		Ok(Some(CreateConsensusState {
+			consensus_state: initial_consensus_state.encode(),
+			consensus_client_id: POLYGON_CONSENSUS_CLIENT_ID,
+			consensus_state_id: self.consensus_state_id,
+			unbonding_period: 60 * 60 * 60 * 27,
```

Comment:
Unbonding period for polygon is 4 days

Reply:
I just deleted this directory.

The current one is https://github.com/polytope-labs/hyperbridge/blob/d04d6a1d31ce1667a73a1044dec192cb2f2f0847/tesseract/consensus/polygon/src/lib.rs#L193